website/addons/twofactor/static/twoFactorUserConfig.js

```diff
@@ -1,63 +1,193 @@
 'use strict';
 
+require('./user-cfg.css');
+
 var ko = require('knockout');
+var Raven = require('raven-js');
 var $ = require('jquery');
-require('./jquery.qrcode.min.js');
+var bootbox = require('bootbox');
+require('jquery-qrcode');
 
 var osfHelpers = require('js/osfHelpers');
+var ChangeMessageMixin = require('js/changeMessage');
 
-var SETTINGS_URL = '/api/v1/settings/twofactor/';
 
-function ViewModel(qrCodeSelector, otpURL) {
+function ViewModel(settingsUrl, qrCodeSelector) {
     var self = this;
+    self.settingsUrl = settingsUrl;
     self.qrCodeSelector = qrCodeSelector;
     self.tfaCode = ko.observable('');
 
     self.message = ko.observable('');
     self.messageClass = ko.observable('');
-    // Initialize QR Code
-    $(self.qrCodeSelector).qrcode(otpURL);
-
-    /** Change the flashed message. */
-    self.changeMessage = function(text, css, timeout) {
-        self.message(text);
-        var cssClass = css || 'text-info';
-        self.messageClass(cssClass);
-        if (timeout) {
-            // Reset message after timeout period
-            setTimeout(function() {
-                self.message('');
-                self.messageClass('text-info');
-            }, timeout);
+
+    self.isEnabled = ko.observable(false);
+    self.isConfirmed = ko.observable(false);
+    self.secret = ko.observable('');
+
+    self.urls = {};
+}
+$.extend(ViewModel.prototype, ChangeMessageMixin.prototype);
```

Comment:
@samanehsan Doesn't the constructor of `ChangeMessageMixin` need to be called in the `ViewModel` constructor?